### PR TITLE
Change credential assertions from String to Struct Logic

### LINKED ISSUE #1198: Change credential assertions from String to Struct Logic
Currently, the credential assertions are String, https://github.com/litentry/litentry-parachain/blob/ccc5abd2bca2d6aedb27f1e824e2460dfe9725fc/tee-worker/litentry/core/credentials/src/lib.rs#L299-L304

they need to be changed to Struct Logic inside Struct Credential

### PR DESCRIPTION
**Pull request by zTgx**
as the title.

### CODE REVIEW

File: tee-worker/litentry/primitives/src/assertion_logic.rs [hunk 1]
```diff
@@ -0,0 +1,143 @@
+// Copyright 2020-2022 Litentry Technologies GmbH.
+// This file is part of Litentry.
+//
```
Comment (DogLi) [inline tee-worker/litentry/primitives/src/assertion_logic.rs]: 2020-2023

Comment (zTgx) [reply tee-worker/litentry/primitives/src/assertion_logic.rs]: updated

File: tee-worker/litentry/primitives/src/assertion_logic.rs [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Litentry.  If not, see <https://www.gnu.org/licenses/>.
+
+// This file includes the predefined rulesets and the corresponding parameters
+// when requesting VCs.
+//
+// See: https://www.notion.so/litentry/Expected-parameters-in-predefined-rulesets-14f74928aa2b43509167da12a3e75507
```
Comment (jingleizhang) [inline tee-worker/litentry/primitives/src/assertion_logic.rs]: This internal link needs to be removed. When external links exist, they should be added back.

File: tee-worker/litentry/primitives/src/lib.rs [hunk 3]
```diff
@@ -22,6 +22,7 @@ mod ethereum_signature;
 mod identity;
 // mod trusted_call;
 mod assertion;
+mod assertion_logic;
```
Comment (jingleizhang) [inline tee-worker/litentry/primitives/src/lib.rs]: Please move this mod to lc-credential

File: tee-worker/litentry/primitives/src/assertion_logic.rs [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+            "#;
+
+		let a1_from_str: AssertionLogic = serde_json::from_str(a1).unwrap();
+
+		let web2_item = AssertionLogic::new_item("$web2_account_cnt", Op::GreaterEq, "7");
+		let web3_item = AssertionLogic::new_item("$web3_account_cnt", Op::GreaterThan, "3");
+
+		let a1_from_struct = AssertionLogic::new_or().add_item(web2_item).add_item(web3_item);
+
+		assert_eq!(a1_from_str, a1_from_struct);
```
Comment (jingleizhang) [inline tee-worker/litentry/primitives/src/assertion_logic.rs]: Please add `assert_eq` about eval() result of a1, while changing "$web2_account_cnt" & "$web3_account_cnt" to actual value

Comment (zTgx) [reply tee-worker/litentry/primitives/src/assertion_logic.rs]: added

File: tee-worker/litentry/primitives/src/lib.rs (hunk 3 above)
Comment (zTgx) [reply tee-worker/litentry/primitives/src/lib.rs]: moved

File: tee-worker/litentry/primitives/src/assertion_logic.rs (hunk 2 above)
Comment (zTgx) [reply tee-worker/litentry/primitives/src/assertion_logic.rs]: ok

**Comment by jingleizhang** [comment]
LGTM, thanks!

**Review by jingleizhang — APPROVED** [review]

**Pull request merged**